include/aws/io/event_loop.h

```diff
@@ -6,286 +6,50 @@
-    void (*destroy)(struct aws_event_loop *event_loop);
-    int (*run)(struct aws_event_loop *event_loop);
-    int (*stop)(struct aws_event_loop *event_loop);
-    int (*wait_for_stop_completion)(struct aws_event_loop *event_loop);
-    void (*schedule_task_now)(struct aws_event_loop *event_loop, struct aws_task *task);
-    void (*schedule_task_future)(struct aws_event_loop *event_loop, struct aws_task *task, uint64_t run_at_nanos);
-    void (*cancel_task)(struct aws_event_loop *event_loop, struct aws_task *task);
-#if AWS_USE_IO_COMPLETION_PORTS
-    int (*connect_to_io_completion_port)(struct aws_event_loop *event_loop, struct aws_io_handle *handle);
-#else
-    int (*subscribe_to_io_events)(
-        struct aws_event_loop *event_loop,
-        struct aws_io_handle *handle,
-        int events,
-        aws_event_loop_on_event_fn *on_event,
-        void *user_data);
-#endif
-    int (*unsubscribe_from_io_events)(struct aws_event_loop *event_loop, struct aws_io_handle *handle);
-    void (*free_io_event_resources)(void *user_data);
-    bool (*is_on_callers_thread)(struct aws_event_loop *event_loop);
-};
-
-struct aws_event_loop {
-    struct aws_event_loop_vtable *vtable;
-    struct aws_allocator *alloc;
-    aws_io_clock_fn *clock;
-    struct aws_hash_table local_data;
-    struct aws_atomic_var current_load_factor;
-    uint64_t latest_tick_start;
-    size_t current_tick_latency_sum;
-    struct aws_atomic_var next_flush_time;
-    void *impl_data;
-};
+struct aws_event_loop_group_options {
 
-struct aws_event_loop_local_object;
-typedef void(aws_event_loop_on_local_object_removed_fn)(struct aws_event_loop_local_object *);
+    /**
+     * How many event loops that event loop group should contain.  For most group types, this implies
+     * the creation and management of an analagous amount of managed threads
+     */
+    uint16_t loop_count;
 
-struct aws_event_loop_local_object {
-    const void *key;
-    void *object;
-    aws_event_loop_on_local_object_removed_fn *on_object_removed;
-};
+    /**
+     * Optional callback to invoke when the event loop group finishes destruction.
+     */
+    struct aws_shutdown_callback_options *shutdown_options;
 
-struct aws_event_loop_options {
-    aws_io_clock_fn *clock;
-    struct aws_thread_options *thread_options;
-};
-
-typedef struct aws_event_loop *(aws_new_event_loop_fn)(struct aws_allocator *alloc,
-                                                       const struct aws_event_loop_options *options,
-                                                       void *new_loop_user_data);
-
-struct aws_event_loop_group {
-    struct aws_allocator *allocator;
-    struct aws_array_list event_loops;
-    struct aws_ref_count ref_count;
-    struct aws_shutdown_callback_options shutdown_options;
+    /**
+     * Optional configuration to control how the event loop group's threads bind to CPU groups
+     */
+    struct aws_event_loop_group_pin_options *pin_options;
```

Comment:
trivial: not a fan of the `aws_event_loop_group_pin_options`, "sub-struct" that contains one int. Binding an additional AwsEventLoopGroupPinOptions class to every language seems like unnecessary work, and unnecessary indirection for users.
```suggestion
    const uint16_t *pin_cpu_group;
```

[Here's an example](https://github.com/awslabs/aws-c-io/blob/e247ef89cdf170cadcb4c665fb0146e82cd6e135/include/aws/io/tls_channel_handler.h#L450-L456) where an optional int was passed by pointer to an options struct, and it worked fine in the bindings ([python](https://github.com/awslabs/aws-crt-python/blob/084e141e9b87aea833b53e5b3a9175320218ed57/awscrt/io.py#L396-L397), [java](https://github.com/awslabs/aws-crt-java/blob/d76fdba766f5e8894be64742d26390ab7223f9f8/src/main/java/software/amazon/awssdk/crt/io/TlsContextPkcs11Options.java#L53))